Resources/Prototypes/Entities/Objects/Fun/Tabletop/mafia.yml

```diff
@@ -0,0 +1,142 @@
+- type: entity
+  name: mafia card box
+  parent: SmallboxItem
+  id: MafiaCardBox
+  description: A box with cards for playing mafia in the amount of 10 players.
+  components:
+  - type: EmitSoundOnLand
+    sound:
+      path: /Audio/Items/matchbox_drop.ogg
+  - type: Sprite
+    netsync: false
+    sprite: Objects\Fun\Tabletop\mafia\mafiacardsbox.rsi
+    state: cardbox
+  - type: Item
+    sprite: Objects\Fun\Tabletop\mafia\mafiacardsbox.rsi
+    size: 10
+  - type: StorageFill
+    contents:
+      - id: CardAgent
+        amount: 2
+      - id: CardDetective
+        amount: 1
+      - id: CardMedic
+        amount: 1
+      - id: CardPassenger
+        amount: 6
+
+- type: entity
+  name: card
+  suffix: Agent
+  parent: BaseItem
+  id: CardAgent
+  description: Turn over the card to find out your role.
+  components:
+  - type: Sprite
+    sprite: Objects\Fun\Tabletop\mafia\mafiacards.rsi
+    drawdepth: SmallObjects
+    layers:
+    - state: cardunknown
+      map: ["enum.StorageVisualLayers.Door"]
+    netsync: false
+  - type: Item
+    sprite: Objects\Fun\Tabletop\mafia\mafiacards.rsi
+    size: 1
+  - type: Storage
+    capacity: -1
+  - type: UserInterface
+    interfaces:
+    - key: enum.StorageUiKey.Key
+      type: StorageBoundUserInterface
+  - type: Appearance
+    visuals:
+    - type: StorageVisualizer
+      state_open: cardagent
+      state_closed: cardunknown
+
+- type: entity
+  name: card
+  suffix: Detective
+  parent: BaseItem
+  id: CardDetective
+  description: Turn over the card to find out your role.
+  components:
+  - type: Sprite
+    sprite: Objects\Fun\Tabletop\mafia\mafiacards.rsi
+    drawdepth: SmallObjects
+    layers:
+    - state: cardunknown
+      map: ["enum.StorageVisualLayers.Door"]
+    netsync: false
+  - type: Item
+    sprite: Objects\Fun\Tabletop\mafia\mafiacards.rsi
+    size: 1
+  - type: Storage
+    capacity: -1
+  - type: UserInterface
+    interfaces:
+    - key: enum.StorageUiKey.Key
+      type: StorageBoundUserInterface
+  - type: Appearance
+    visuals:
+    - type: StorageVisualizer
+      state_open: carddetective
+      state_closed: cardunknown
+
+- type: entity
+  name: card
+  suffix: Passenger
+  parent: BaseItem
+  id: CardPassenger
+  description: Turn over the card to find out your role.
+  components:
+  - type: Sprite
+    sprite: Objects\Fun\Tabletop\mafia\mafiacards.rsi
+    drawdepth: SmallObjects
+    layers:
+    - state: cardunknown
+      map: ["enum.StorageVisualLayers.Door"]
+    netsync: false
+  - type: Item
+    sprite: Objects\Fun\Tabletop\mafia\mafiacards.rsi
+    size: 1
+  - type: Storage
+    capacity: -1
```

Comment:
this is a funny hack, is there any other existing way to visualize it?

Reply:
> this is a funny hack, is there any other existing way to visualize it?

I failed, at best it stopped spawning, and at worst it restarting the server.

Reply:
> as in removing all the storage visualizer/ui related stuff and using something intended for flipping cards

I didn't find anything like that, the closest was SpawnOnUse, but the cards stopped spawning. To make everything beautiful, you need to create a new interface via #C, but this is beyond my capabilities.

Reply:
ok keep the funny hack for now i guess, if it still exists when merged i will make write a visualizer to toggle sprite state